api/client/lib.py

```diff
@@ -528,13 +458,15 @@ def list_of_series_to_single_series(series_list, add_belongs_to=False, include_h
         if not (isinstance(series, dict) and isinstance(series.get('data', []), list)):
             continue
         series_metadata = series.get('series', {}).get('metadata', {})
-        if not include_historical and (series_metadata.get('includesHistoricalRegion', False) or 
-        series_metadata.get('includesHistoricalPartnerRegion', False)):
+        has_historical_regions = (series_metadata.get('includesHistoricalRegion', False) or
+                                  series_metadata.get('includesHistoricalPartnerRegion', False))
+        if not include_historical and has_historical_regions:
             continue
         # All the belongsTo keys are in camelCase. Convert them to snake_case.
         # Only need to do this once per series, so do this outside of the list
         # comprehension and save to a variable to avoid duplicate work:
-        belongs_to = camel_to_snake_dict(series.get('series', {}).get('belongsTo', {}))
+        belongs_to = dict_reformat_keys(series.get('series', {}).get('belongsTo', {}),
+                                        str_camel_to_snake)
```

Comment:
getting error here `str_camel_to_snake() takes 1 positional argument but 2 were given`

Reply:
Should be fixed now. Fantastic save - there was a paren out of place. Instead of

```py
dict(format_func((key), value) for key, value in obj.items())
```

it was supposed to be

```py
dict((format_func(key), value) for key, value in obj.items())
```

I was very surprised that the unit test here https://github.com/gro-intelligence/api-client/blob/development/api/client/lib_test.py#L149 didn't catch that, but it's just because the `series` object in that test has no keys. So this loop runs 0 times and that function's never called.

I have now:

1. re-factored that line to not have so many parens so it's more readable https://github.com/gro-intelligence/api-client/pull/161/commits/b5478d0519864dd14c477382da79331d995ba16f
2. added the doctests to shippable. The test was failing for that function, but shippable's never been automatically running the doctests until now https://github.com/gro-intelligence/api-client/pull/161/commits/637e43284b003d8c3ef2039729b334d99320bc56